code/graphics/2d.cpp

```diff
@@ -1597,6 +1597,14 @@ bool gr_init(std::unique_ptr<os::GraphicsOperations>&& graphicsOps, int d_mode,
 	mprintf(("  Persistent buffer mapping: %s\n",
 	         gr_is_capable(CAPABILITY_PERSISTENT_BUFFER_MAPPING) ? "Enabled" : "Disabled"));
 
+	mprintf(("Checking mod required rendering features...\n"));
+	for (int ext : Required_render_ext) {
+		if (!gr_is_capable(static_cast<gr_capability>(ext))) {
+			Error(LOCATION, "  Feature %d required by mod not supported by system.\n", ext);
```

Comment:
Since this is a user facing error message it would be good to have a textual representation of the feature that is missing. Also, why are there two leading spaces and a `\n` at the end?

Reply:
The two leading spaces is pretty standard formatting across the `gr_init` calls to mprintf (look a few lines higher for example).

Reply:
Yes, but this isn't an `mprintf` call.

Reply:
Alright, I'll change it. Also, where would be the proper place to put a helper function?

Reply:
Probably here in 2d.cpp. The enum is also defined here so that makes sense to me.